src/ansys/geometry/core/misc/checks.py

```diff
@@ -331,19 +331,19 @@ def min_backend_version(major: int, minor: int, service_pack: int):
 
     def backend_version_decorator(method):
         def wrapper(self, *args, **kwargs):
-            method_version = f"{major}.{minor}.{service_pack}"
+            method_version = semver.Version(major, minor, service_pack)
             if hasattr(self, "_grpc_client"):
                 if self._grpc_client is None:
                     raise GeometryRuntimeError(
                         "The client is not available. You must initialize the client first."
                     )
                 elif self._grpc_client.backend_version is not None:
-                    comp = semver.compare(method_version, self._grpc_client.backend_version)
+                    comp = method_version.compare(self._grpc_client.backend_version)
                     # if comp is 1, method version is higher than backend version.
                     if comp == 1:
 
                         # Check if the version is "0.0.0" (i.e., the version is not available)
-                        if self._grpc_client.backend_version == "0.0.0":
+                        if str(self._grpc_client.backend_version) == "0.0.0":
```

Comment:
Nitpicking, maybe is better to use the semver package directly to compare instead of strings? I couldn't find an exact version comparison, but we can use this. It is a bit more verbose, up to you:
```suggestion
                        if self._grpc_client.backend_version.is_compatible(semver.Version(0, 0, 0)):
```

Reply:
Hmm got your point... but I'd prefer to avoid using the Version object there... more time consuming plus it is an edge case honestly (0.0.0 shouldn't happen starting on 24R2... but it might happen A LOT in older versions - i.e. 24R1). The string representation of the object you get it for free - so since it is such an overwhelmingly called method, I'd avoid doing that. Thanks for the suggestion nonetheless!